src/Component.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Processor\FormatCsv;
+
+use Exception;
+use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Symfony\Component\Finder\Finder;
+use Throwable;
+use function str_replace;
+
+class Component extends BaseComponent
+{
+    public function run(): void
+    {
+        /** @var Config $config */
+        $config = $this->getConfig();
+
+        $tablesFinder = new Finder();
+        $tablesFinder
+            ->in($this->getDataDir() . "/in/tables")
+            ->files();
+
+        foreach ($tablesFinder as $csvTableFrom) {
+            try {
+                $tableManifest = $this->getManifestManager()->getTableManifest($csvTableFrom->getFilename());
+                if (!isset($tableManifest['delimiter'], $tableManifest['enclosure'])) {
+                    throw new Exception('Table manifest must contain delimiter and enclosure.');
+                }
+            } catch (Throwable $e) {
```

Comment:
Nemůže se stát že to vyhodí něco co není user error?

Reply:
No, bohužel jsem se tam nachytal na to, že jsem tam dal moc generickou exceptionu :'( Musel bych to upravit v upstreamu v php-component. Může to vyhodit `\InvalidArgumentException` nebo `\Exception`. Asi bych to nechal, i když souhlasím, že je to špatně. Udělám issue na to abych to opravil později.

Reply:
#3

Reply:
ok, souhlasím s issue a opravou později. 
Díval jsem se co to tam volá a v podstatě to jenom čte soubor takže tam toho moc nehrozí. I když jeden čas jsme měli problémy s přetížením EBS disků kdy docházelo k chybám právě při čtení souborů a v takovém případě by nám to problém mohlo zatajit a uživateli naopak říct zavádějící hlášku....

Reply:
Tak jako tak jsem to lehce poladil, aby to líp chytalo. Z toho čtení to chytá jen invalid argument exception a když v manifestu chybí delimiter/encapsulator, tak to vyhodí přímo userexception

Reply:
Blbost, opravil jsem to špatně. Tak znovu.

Reply:
Otestoval jsem si to lokálně smazáním manifestu. Už je to OK